### PR TITLE
Add EventLogging option

### LINKED ISSUE #938: make event logging configurable
### What version of Go are you using (`go version`)?

<pre>
$ go version
1.12.7
</pre>

### What version of Badger are you using?
v1.6.0

### What did you do?

Ran benchmarks using badgerdb.

### What did you expect to see?

I *didn't* expect to see logging appear on the CPU profile.

### What did you see instead?

It did, and there's no way to turn it off.

In some of my benchmarks ~10% of the time spent inside (*DB).doWrites (and more elsewhere!) is spent printing to the event log. ISTM there's no way to disable this.

### PR DESCRIPTION
**Pull request by mjgarton**
This introduces an option to disable or enable trace.EventLog logging of
events.  For backwards compatability this defaults to true.

The reason for making this configurable is that under some workloads the
cost of event logging is significant and so can cause performance
issues.

Fixes #938

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/badger/1035)

### CODE REVIEW

File: nil_event_log.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package badger
```
[the remark this answers is not in the document]
Comment (ashish-goswami) [reply nil_event_log.go]: We can call this file just `event_log.go`.

**Comment by ashish-goswami** [comment]
@mjgarton We need to propagate above change https://github.com/dgraph-io/badger/blob/86a77bb28c44052be16ce878ca1115ec0b83824e/value.go#L795 here also.

File: value.go [hunk 2]
```diff
@@ -787,12 +787,15 @@ func (vlog *valueLog) replayLog(lf *logFile, offset uint32, replayFn logEntry) e
 	return nil
 }
 
-func (vlog *valueLog) open(db *DB, ptr valuePointer, replayFn logEntry) error {
+func (vlog *valueLog) open(db *DB, ptr valuePointer, replayFn logEntry, eventLogging bool) error {
```
Comment (ashish-goswami) [inline value.go]: No need to pass extra argument. It should be accessible from from `db.opt`.

**Comment by mjgarton** [comment]
I missed another event logger in `y/watermark.go` so I need to add that.  I'll push some changes soon.

**Comment by mjgarton** [comment]
Please advise whether `y` was the right place to put the null event logging code.  It isn't quite clear to me what the scope of the `y` package is supposed to be and I didn't find it documented.

File: y/event_log.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	NoEventLog trace.EventLog = nilEventLog{}
+)
+
+type nilEventLog struct{}
+
+func (nel nilEventLog) Printf(format string, a ...interface{}) {}
+
+func (nel nilEventLog) Errorf(format string, a ...interface{}) {}
+
+func (nel nilEventLog) Finish() {}
```
Comment (ashish-goswami) [inline y/event_log.go]: You might want to implement `SetMaxEvents` function also. As it is used below.
https://github.com/dgraph-io/badger/blob/c6681c181835f3846b859d6b546e5da8d9f86ea9/levels.go#L845

**Comment by jarifibrahim** [comment]
Thank you for fixing this @mjgarton :tada:

**Pull request merged**